packages/jaeger-ui/src/components/common/UiFindInput.tsx

```diff
@@ -77,16 +83,33 @@ export class UnconnectedUiFindInput extends React.PureComponent<TProps, StateTyp
   };
 
   render() {
+    const { allowClear, forwardedRef, inputProps } = this.props;
+
     const inputValue = _isString(this.state.ownInputValue) ? this.state.ownInputValue : this.props.uiFind;
 
+    // antd cannot handle suffix changing from present to absent or vice versa
+    let suffix: React.ReactElement = inputProps.suffix || <></>;
+    if (inputValue && inputValue.length && allowClear) {
+      suffix = <Icon type="close" onClick={this.clearUiFind} />;
+      if (inputProps.suffix) {
+        suffix = (
+          <>
+            {suffix}
+            {inputProps.suffix}
+          </>
+        );
+      }
+    }
```

Comment:
I based it off of [this in plexus](https://github.com/jaegertracing/jaeger-ui/blob/master/packages/plexus/src/LayeredDigraph/SvgLayer.tsx#L71)
would this:
```
let content = input.suffix || <></>;
if (allowClear && inputValue && inputValue.length) {
  content = (
    <>
      <Icon type="close" onClick={this.clearUiFind} />
      {content}
    </>
  );
}
```
be clearer?
It's got less branching logic, but results in an empty `React.fragment` in more cases. Previously it was just when both `input.suffix` is absent and `allowClear` is falsy, now it is whenever `input.suffix` is absent. though extraneous empty fragments shouldn't have an impact so that probably doesn't matter

Reply:
Would this work?

```tsx
const suffix = (
  <>
    {inputValue && inputValue.length && allowClear && (
      <Icon type="close" onClick={this.clearUiFind} />
    )}
    {inputProps.suffix}
  </>
);
```

Reply:
ah yeah that is much clearer